### PR TITLE
Throw exception when asked for missing state

### PR DESCRIPTION
**Pull request by walterdolce**
Implementation for #18.

Note the change in `hasStateFragment`. This is because PHP might give you a `false` even though you actually have the key set. Please see https://eval.in/678234.

### CODE REVIEW

File: src/ScenarioState.php [hunk 1]
```diff
@@ -26,14 +28,17 @@ class ScenarioState implements ScenarioStateInterface
...
     }
 
     /**
      * {@inheritdoc}
      */
     public function getStateFragment($key)
     {
+        if (!$this->hasStateFragment($key)) {
+            throw new MissingStateException("Missing {$key} state fragment was requested from store.");
+        }
```
Comment (vincentchalamon) [inline src/ScenarioState.php:41]: Break line before `return` instance.

File: src/ScenarioState/Exception/MissingStateException.php [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * (c) Rodrigue Villetard <rodrigue.villetard@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace Gorghoa\ScenarioStateBehatExtension\ScenarioState\Exception;
+
+/**
+ * Class MissingStateException
```
Comment (vincentchalamon) [inline src/ScenarioState/Exception/MissingStateException.php]: Useless description

File: src/ScenarioState/Exception/MissingStateException.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * @package Gorghoa\ScenarioStateBehatExtension\ScenarioState\Exception
```
Comment (vincentchalamon) [inline src/ScenarioState/Exception/MissingStateException.php]: Useless `@package` annotation.
Can you also add a `@author` annotation with your name please?

File: src/ScenarioState/Exception/MissingStateException.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ */
+class MissingStateException extends \Exception
```
Comment (vincentchalamon) [inline src/ScenarioState/Exception/MissingStateException.php]: Prefer extends `\LogicException`: http://php.net/manual/en/class.logicexception.php

File: tests/ScenarioStateTest.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+namespace Gorghoa\ScenarioStateBehatExtension;
+
+use Gorghoa\ScenarioStateBehatExtension\ScenarioState\Exception\MissingStateException;
+
+/**
+ * Class ScenarioStateTest
```
Comment (vincentchalamon) [inline tests/ScenarioStateTest.php]: Useless description

File: tests/ScenarioStateTest.php [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ * @package Gorghoa\ScenarioStateBehatExtension
```
Comment (vincentchalamon) [inline tests/ScenarioStateTest.php]: Useless `@package` annotation
Can you add a `@author` annotation with your name please?

**Review by vincentchalamon — CHANGES_REQUESTED** [review]
One main change required, and some minor ones

File: src/ScenarioState.php (hunk 1 above)
Comment (walterdolce) [reply src/ScenarioState.php:41]: Done!

File: src/ScenarioState/Exception/MissingStateException.php (hunk 2 above)
Comment (walterdolce) [reply src/ScenarioState/Exception/MissingStateException.php]: Done!

File: src/ScenarioState/Exception/MissingStateException.php (hunk 3 above)
Comment (walterdolce) [reply src/ScenarioState/Exception/MissingStateException.php]: Done!

File: src/ScenarioState/Exception/MissingStateException.php (hunk 4 above)
Comment (walterdolce) [reply src/ScenarioState/Exception/MissingStateException.php]: Done!

File: tests/ScenarioStateTest.php (hunk 5 above)
Comment (walterdolce) [reply tests/ScenarioStateTest.php]: Done!

File: tests/ScenarioStateTest.php (hunk 6 above)
Comment (walterdolce) [reply tests/ScenarioStateTest.php]: Done!

**Review by vincentchalamon — APPROVED** [review]

**Comment by vincentchalamon** [comment]
Thanks @walterdolce, I'll release it later as it's not blocking as previous PRs

**Pull request merged**